content/concepts/hot-module-replacement.md

```diff
@@ -6,52 +6,87 @@ contributors:
+between these builds. It stores these IDs in a "records" JSON file or in memory.
 
 ### From The Module View
 
-HMR is an opt-in feature that affects only modules containing HMR code. You can consider patching styling through style-loader as a good example. To make this work, style-loader implements the HMR interface. Implementing it you describe what should happen when a module is updated. In this case we simply replace the styling with the one received through HMR.
+HMR is an opt-in feature that only affects modules containing HMR code. One example
+would be patching styling through the [style-loder](https://github.com/webpack/style-loader).
+In order for patching to work, style-loader implements the HMR interface; when it
+receives an update through HMR, it replaces the old styles with the new ones.
 
-In most cases it's not mandatory to write HMR code in every module. If a module has no HMR handlers the update bubbles up. This means a single handler can handle an update to a complete module tree. If a single module in this tree is updated, the complete module tree is reloaded (only reloaded not transferred).
+Similarly, when implementing the HMR interface in a module, you can describe what should
+happen when the module is updated. However, in most cases, it's not mandatory to write
+HMR code in every module. If a module has no HMR handlers, the update bubbles up. This
+means that a single handler can handle an update to a complete module tree. If a single
+module in this tree is updated, the complete module tree is reloaded (only reloaded,
+not transferred).
 
 ### From The HMR Runtime View (Technical)
 
-For the module system runtime additional code is emitted to track module `parents` and `children`.
-
-On the management side the runtime supports two methods: `check` and `apply`.
+For the module system runtime, additional code is emitted to track module `parents` and `children`.
 
-A `check` does a HTTP request to the update manifest. When this request fails, there is no update available. Elsewise the list of updated chunks is compared to the list of currently loaded chunks. For each loaded chunk the corresponding update chunk is downloaded. All module updates as stored in the runtime as update. The runtime switches into the `ready` state, meaning an update has been downloaded and is ready to be applied.
+On the management side, the runtime supports two methods: `check` and `apply`.
 
-For each new chunk request in the ready state the update chunk is also downloaded.
+A `check` makes an HTTP request to the update manifest. If this request fails,
+there is no update available. If it succeeds, the list of updated chunks is compared
+to the list of currently loaded chunks. For each loaded chunk, the corresponding
+update chunk is downloaded. All module updates are stored in the runtime.
+When all update chunks have been downloaded and are ready to be applied, the runtime
```

Comment:
Is this correct? The existing sentence was unclear, so I went to the source code, but I'm not 100% sure.